pipeline/config.go

```diff
@@ -21,17 +21,25 @@ import (
 // Config is the configuration of a pipeline.
 type Config []operator.Config
 
-// BuildOperators builds the operators from the list of configs into operators
+// BuildOperators builds the operators from the list of configs into operators.
 func (c Config) BuildOperators(bc operator.BuildContext) ([]operator.Operator, error) {
 	operators := make([]operator.Operator, 0, len(c))
-	for i, builder := range c {
-		nbc := getBuildContextWithDefaultOutput(c, i, bc)
-		op, err := builder.Build(nbc)
+	// buildsMulti is used for storing a key of the Builder that builds multiple operators.
+	// The map is then used in SetOutputIDs to assign the output value of any ops pointing to the Builder as their output to the first operator in said Builder.
+	buildsMulti := make(map[string]string)
```

Comment:
I find this explanation and variable name confusing. What does the key represent? What does the value represent?

Reply:
If I understand correctly, the key is the ID of a plugin, and the value is the ID of that plugin's first operator. Is that right?

Reply:
yes, I was trying to generalize it in case we add operators that build multiple ops as well.